### PR TITLE
inline variables in format && to_xx_case_ident && field_arrays

### PR DESCRIPTION
**Pull request by burrbull**
r? @Emilgardis

### CODE REVIEW

**Review by Emilgardis — DISMISSED** [review]

File: ci/svd2rust-regress/src/tests.rs [hunk 1]
```diff
@@ -81,94 +84,51 @@ use self::RunWhen::*;
 const BLACKLIST_CHARS: &[char] = &['(', ')', '[', ']'];
 
 /// Lovingly stolen from `svd2rust`. Probably could be `Cow`
-pub trait ToSanitizedSnakeCase {
-    fn to_sanitized_snake_case(&self) -> String;
+pub trait ToSanitizedCase {
+    fn to_sanitized_not_keyword_snake_case(&self) -> Cow<str>;
+    fn to_sanitized_snake_case(&self) -> Cow<str> {
```
[the remark this answers is not in the document]
Comment (burrbull) [reply ci/svd2rust-regress/src/tests.rs]: done

**Comment by burrbull** [comment]
cc @Emilgardis

**Comment by burrbull** [comment]
Yet one old bug :)

**Comment by burrbull** [comment]
Fixes #639
сс @duskmoon314

**Comment by duskmoon314** [comment]
LGTM

**Review by therealprof — APPROVED** [review]
LGTM, thanks

bors r+

**Comment by burrbull** [comment]
bors retry

**Comment by burrbull** [comment]
bors retry

**Pull request merged**